orangecontrib/text/corpus.py

```diff
@@ -53,6 +53,18 @@ def _check_arrays(*arrays):
     return lengths.pop() if len(lengths) else 0
 
 
+@contextmanager
+def corpora_dataset_dir():
```

Comment:
Not thread safe. :) Nit picking, I know, but there probably are other solutions.

Reply:
You are right. I currently do not see a good solution. Maybe we can move the `dataset_dirs` table in the Table class in Orange and override it in Corpus. It would be nicer. What do you think about that solution?

Reply:
I like what you suggest. Another possible implementation (for now) would be to reimplement (copy) the file finding functionality into Corpus.

Reply:
I used similar file fining functionality that was used before.